crates/polars-time/src/chunkedarray/rolling_window/dispatch.rs

```diff
@@ -195,7 +195,13 @@ pub trait SeriesOpsTime: AsSeries {
         by: &Series,
         options: RollingOptionsDynamicWindow,
     ) -> PolarsResult<Series> {
-        let s = self.as_series().clone();
+        let mut s = self.as_series().clone();
+        if matches!(
+            s.dtype(),
+            DataType::Int8 | DataType::UInt8 | DataType::Int16 | DataType::UInt16
+        ) {
+            s = s.cast(&DataType::Int64).unwrap();
```

Comment:
It be nice if this cast could be promoted into the logical plan type coercion phase so that we don't have to mimic this behaviour in the gpu engine too.